tornado/test/websocket_test.py

```diff
@@ -329,6 +336,13 @@ def test_check_origin_invalid_subdomains(self):
 
         self.assertEqual(cm.exception.code, 403)
 
+    @gen_test
+    def test_handshake_response_headers(self):
+        ws = yield self.ws_connect('/handshake_response_headers')
+
+        self.assertEqual(ws.headers.get('Set-Cookie'), 'hello=1')
```

Comment:
At least it works in latest Chrome and Firefox on OS X - I've tested there. But as I see from cookies RFC 6265 - browsers are free to ignore it: "User agents MAY ignore Set-Cookie headers contained in responses with 100-level status codes"